source/client_channel_handler.c

```diff
@@ -817,6 +848,9 @@ static void s_request_outgoing_task(struct aws_channel_task *task, void *arg, en
                 mqtt_connection_unlock_synced_data(connection);
             } /* END CRITICAL SECTION */
 
+            /* Since a request has complete, update the next ping time */
+            s_update_next_ping_time(connection);
```

Comment:
Let's move this into the critical section above.  Update references channel/slot which could be side-affected under us if we're not in the event loop thread.